### ISSUE TITLE
End-to-end Neural Coreference Resolution (EMNLP 2017)

### ISSUE THREAD

**Issue by iBotamon** [yutanakamura-tky/paper_reading#13]
https://www.aclweb.org/anthology/D17-1018.pdf

# どんなもの？
    - 共参照解析をはじめて end-to-end で行った研究.
    - つまり, entityの同定と共参照解析を同時に行う.
# 先行研究と比べてどこがすごい？
# 技術や手法のキモはどこ？
    - 1. 問題の定式化
        - T個のトークンからなる文書Dに対して共参照解析を行うとする.
        - 文書Dから長さ 1,...,T の部分トークン列を N=T(T+1)/2 通り取り出すことができる. 個々の部分トークン列を span と呼ぶことにする.
        - N個の span を開始位置(と, 終了位置)が若い順にソートし, 順に 1,...,N の番号を振る.
        - 共参照解析は次のことを行うことに相当する:
            - span i に対し, 共参照のペアとなる span yi を割り当てる.
            - ただし span yi は span i より番号が若いとする.
            - また, 共参照のペアとなるべき span が存在しない, もしくは span i がそもそもentityでない場合は, yi = ε とする.
            - 上記をまとめて, yi の候補を Yi = {ε, 1, ..., i-1} とおく.
            - このような yi の推定を i = 1,...,N に対して行う.
    - 2. 事後確率の定式化
        - yiの事後確率を P(yi|D) とおく.
        - そうすると, 共参照解析は, 同時確率 logP(y1,...,yN|D) を最大化するような y1,...,yN を求めることにより実行できる.
        - ここでyiの事後確率は, Yi中の全候補に対して確からしさのscoreを求め, そのsoftmaxをとったものであるとする.
        - yi = j となる確からしさのscoreを s(i,j) とおく. s(i,j) は次のように定義する:
            - j = ε ならば, s(i,j) = 0
            - そうでなければ, s(i,j) = sm(i) + sm(j) + sa(i,j)
                - ただし sm(i) は span i が entity でありそうな度合い.
                - ただし sm(j) は span j が entity でありそうな度合い.
                - ただし sa(i,j) は span j が span i の共参照のペアとなってそうな度合い.
        - 以上をまとめると, y1,...,yN = argmax sum(log(softmax(score(yi|D) over yi in Yi)))
    - 3. Span representation と score
        - sm(), sa() は次のように計算する:
            - sm(i) = dot(wm, FFNN_m(gi))
            - sa(i,j) = dot(wa, FFNN_a(concat(gi, gj, hadamar(gi,gj), φ(i,j))))
            - ここで gi, gj は span i, j に対する representation(後述), wm, wa は重みベクトル, FFNNはfeed-forward network.
            - また, φ(i,j)はspan i,jから得られる特徴量であって, 次のような特徴量ベクトルをconcatしたものであるとする:
                - i, j の距離に対する20次元のembedding. ただし距離は1,2,3,4,5-7,8-15,16-31,32-63,64+で量子化する.
                - i, j が同一の話者によって話されているかのbool値に対する20次元のembedding.
                - i, j のジャンル(?)に対する20次元のembedding.
                - i, j の長さそれぞれに対する20次元のembedding. ただし後述するが, span長には実際には上限を設けているため, 上限をLとするとspan長は1,...,LのL種類だけ考慮すれば良い.
    - 4. Span representation の取得
        - まず文Dの各トークンを token 1, ..., token T とおく.
        - token t に対し, GloVeの300次元ベクトル, Turianの50次元ベクトル, 文字単位CNNに通して得る8次元ベクトルをconcatし, これを xt とおく.
            - Glove, TurianについてはOOVの場合は零ベクトルで代用する.
        - x1, ..., xT を BiLSTM に通す. xt の双方向の隠れ状態をconcatしたものを xt\* とおく.
        - span i の開始位置, 終了位置を Start(i), End(i) とおき, Start(i) <= j <= End(i) に対する xj の重み付き平均を xi^ とする.
            - xj^ はattentionによって計算する.
            - αt = dot(wα, FFNN_α(xt\*)) とする. wαは重みベクトル.
            - 次に, xj^ を x_Start(i), ..., x_End(i) の重み付き平均によって得る. 重みは αt を span区間にわたって softmax で正規化して得る.
        - さらに, φ(i) を span i の長さに対する20次元の embedding によって得る (前節参照)
        - 最後に x_Start(i)\*, x_End(i)\*, xi^, φ(i) を concat したものを gi とする. これが span representation.
    - 5. 推論時の計算量を減らすために
        - 次のような制約を設ける:
            - span は最長でLトークンまでとする.
            - sm() の計算の結果, 高いλT個の span だけを保持し, 残りは破棄する. つまりentityは最大でもλT個までしか認識しない.
            - 上記の λT個の span 候補は, sm() の高い順に "accept" していく. ただし, すでにacceptされた他のspan候補と区間がoverlapする場合はrejectする.
            - acceptされたspanそれぞれに対し, 共参照のペアの候補は上位K個だけ保持し, 残りは破棄する.
    - 6. Loss function
        - sm, sa は Loss には組み込まず, yi の事後確率を単に最適化している.
# どうやって有効だと検証した？
    - データセット
        - CoNLL 2012 shared task
            - OntoNotesから作成したデータ.
    - 評価指標
        - CoNLL 2012に従う: precision, recall, F1, MUC, B^3, CEAF_φ4
    - ベースライン
        - 従来手法9種類.
    - 結果
        - 当時のSOTAを更新.
# 議論はある？
# 次に読むべき論文は？
